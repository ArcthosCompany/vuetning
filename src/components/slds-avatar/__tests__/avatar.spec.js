import {mount} from '@vue/test-utils'
import SldsAvatar from '../avatar.vue'

describe('SldsAvatar', () => {
    const src = 'https://s3-sa-east-1.amazonaws.com/bluebear.app/logo.svg'
    const initials = 'MG'
    const fallbackIcon = 'standard:account'
    const alternativeText = 'alternative text'

    describe('Image', () => {

        it('should not render when invalid', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {src: null, fallbackIcon}
            })

            expect(wrapper.find('img').exists()).toBeFalsy()
        })

        it('should render when valid', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {src}
            })

            expect(wrapper.find('img').exists()).toBeTruthy()
        })

        it('should correct image src when changed', async () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {src}
            })

            expect(wrapper.element).toMatchSnapshot()
            await wrapper.setProps({src: 'https://s3-sa-east-1.amazonaws.com/bluebear.app/logo-name.png'})
            expect(wrapper.element).toMatchSnapshot()
        })

        it('should render alternative text when set', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {src, alternativeText}
            })

            expect(wrapper.element).toMatchSnapshot()
        })

        it('should correct alternative text when changed', async () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {alternativeText}
            })

            expect(wrapper.element).toMatchSnapshot()
            await wrapper.setProps({alternativeText: 'new alternative text'})
            expect(wrapper.element).toMatchSnapshot()
        })

    })

    describe('Initials', () => {

        it('should not render when invalid', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {initials: null, fallbackIcon}
            })

            expect(wrapper.find('abbr').exists()).toBeFalsy()
        })

        it('should render when valid', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {initials, fallbackIcon}
            })

            expect(wrapper.find('abbr').exists()).toBeTruthy()
        })

        it('should render class slds-icon-standard-account when fallbackIconName is standard:account', () => {
            const wrapper = mount(SldsAvatar, {
                propsData: {initials, fallbackIcon}
            })

            expect(wrapper.find('.slds-icon-standard-account').exists()).toBeTruthy()
        })

    })

    describe('Variant', () => {

    })

    describe('Size', () => {

    })
})
