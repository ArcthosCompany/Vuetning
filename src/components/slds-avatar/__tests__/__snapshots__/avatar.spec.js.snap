// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SldsAvatar Image should correct alternative text when changed 1`] = `
<span
  class="slds-avatar slds-avatar_medium"
>
  <!-- Image -->
  
  <!-- Fallback icon -->
  <span
    alternative-text="alternative text"
    class="slds-icon_container"
  >
    <slds-svg
      aria-hidden="true"
      class="slds-icon slds-icon_medium"
      icon="standard:user"
      standard-format="false"
    />
    <!--v-if-->
  </span>
  
</span>
`;

exports[`SldsAvatar Image should correct alternative text when changed 2`] = `
<span
  class="slds-avatar slds-avatar_medium"
>
  <!-- Image -->
  
  <!-- Fallback icon -->
  <span
    alternative-text="new alternative text"
    class="slds-icon_container slds-icon-standard-user"
  >
    <slds-svg
      aria-hidden="true"
      class="slds-icon slds-icon_medium"
      icon="standard:user"
      standard-format="false"
    />
    <!--v-if-->
  </span>
  
</span>
`;

exports[`SldsAvatar Image should correct image src when changed 1`] = `
<span
  class="slds-avatar slds-avatar_medium"
>
  <!-- Image -->
  <img
    src="https://s3-sa-east-1.amazonaws.com/bluebear.app/logo.svg"
  />
</span>
`;

exports[`SldsAvatar Image should correct image src when changed 2`] = `
<span
  class="slds-avatar slds-avatar_medium"
>
  <!-- Image -->
  <img
    src="https://s3-sa-east-1.amazonaws.com/bluebear.app/logo-name.png"
  />
</span>
`;

exports[`SldsAvatar Image should render alternative text when set 1`] = `
<span
  class="slds-avatar slds-avatar_medium"
>
  <!-- Image -->
  <img
    alt="alternative text"
    src="https://s3-sa-east-1.amazonaws.com/bluebear.app/logo.svg"
  />
</span>
`;
